lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import * as rtlUtils from '../../../__testutils__/rtlUtils';
+
+import userEvent from '@testing-library/user-event';
+
+import NxCombobox, {Props} from '../NxCombobox';
+
+describe('NxCombobox', function() {
+  const minimalProps: Props<string | number> = {
+        value: '',
+        onChange: () => {},
+        onSearch: () => {},
+        matches: []
+      },
+      rtlRender = rtlUtils.rtlRender(NxCombobox, minimalProps);
+
+  //mock scrollIntoView function using jest
+  window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+  it('renders a div with the nx-combobox class and the specified attributes', function() {
+    const { container } = rtlRender({ title: 'bar' }),
+        component = container.firstElementChild!;
+
+    expect(component.tagName).toBe('DIV');
+    expect(component).toHaveClass('nx-combobox');
+    expect(component).toHaveAttribute('title', 'bar');
+  });
+
+  it('adds the ref to the div', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        domNode = render(<NxCombobox { ...minimalProps } ref={ref} />).container.firstElementChild!;
+
+    expect(ref.current).toBeDefined();
+    expect(ref.current).toBe(domNode);
+  });
+
+  it('renders an input child with the combobox role', function() {
+    const { getByRole } = rtlRender(),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement.tagName).toBe('INPUT');
+    expect(inputElement).toHaveAttribute('role', 'combobox');
+  });
+
+  it('sets the value as the value of the input', function() {
+    const { getByRole } = rtlRender({ value: 'foo' }),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement).toHaveValue('foo');
+  });
+
+  it('sets aria-autocomplete on the input to list as default and to both if `autoComplete` prop is set to true',
+      function() {
+        const { getByRole, rerender } = rtlRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox {...minimalProps} autoComplete={true} />);
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const onChange = jest.fn(),
+            { getByRole } = rtlRender({ autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }],
+              onChange}),
+            inputElement = getByRole('combobox');
+
+        inputElement.focus();
+        await userEvent.type(inputElement, 'f');
+        expect(onChange).toBeCalledWith('Foo');
+      });
+
+  it('sets aria-expanded on the input to true when the dropdown displayed which has aria-hidden set to false',
```

Comment:
Should we use `aria-hidden` or would it be simpler to just not render the dropdown at all when it's closed?

Reply:
As we discussed in slack, will have it offscreen along with aria-hidden. Thanks.